staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/fake_queuewait.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"context"
+	"time"
+
+	testingpromise "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/promise"
+)
+
+func NewFakeQueueWait() fakeQueueWait {
+	return fakeQueueWait{}
+}
+
+type fakeQueueWait struct{}
```

Comment:
It would be good to declare the interface that this type is intended to implement.